lib/Translation/ImporterService/Importer/DataObjectImporter.php

```diff
@@ -69,14 +69,20 @@ protected function importAttribute(Element\ElementInterface $element, string $ta
 
             /** @var array $blockData */
             $blockData = $element->{'get' . $blockName}($targetLanguage);
-            $blockItem = !empty($blockData) && $blockData[$blockIndex] ? $blockData[$blockIndex] : clone $originalBlockItem;
+            // Convert to object for cloning
+            if(!is_object ($originalBlockItem) ) {
+                $originalBlockItem = (object)$originalBlockItem;
+            }
 
+            $blockItem =  !empty($blockData) && isset($blockData[$blockIndex]) ? $blockData[$blockIndex] : clone $originalBlockItem;
+            $blockItem = (array)$blockItem;
```

Comment:
```suggestion
            $blockItem =  isset($blockData[$blockIndex]) ? $blockData[$blockIndex] : $originalBlockItem;
```

this would be enough, it's always an array (even with block max items 1)